xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -340,111 +328,102 @@ public boolean isEquivalentTo(RoundRobinPicker picker) {
-   *
-   * <p>Specifically, each object added to chooser is given a deadline equal to the multiplicative
-   * inverse of its weight. The place of each object in its deadline is tracked, and each call to
-   * choose returns the object with the least remaining time in its deadline.
-   * (Ties are broken by the order in which the children were added to the chooser.) The deadline
-   * advances by the multiplicative inverse of the object's weight.
-   * For example, if items A and B are added with weights 0.5 and 0.2, successive chooses return:
+  /*
+   * Implementation of Static Stride Scheduler, replaces EDFScheduler.
+   * <p>
+   * The Static Stride Scheduler works by iterating through the list of subchannel weights
+   * and using modular arithmetic to proportionally distribute picks, favoring entries 
+   * with higher weights. It is based on the observation that the intended sequence generated 
+   * from the EDF scheduler is a periodic one that can be achieved through modular arithmetic. 
+   * This scheduler generates a practically equivalent sequence of picks as the EDFScheduler.
+   * The Static Stride Scheduler is more performant than the EDFScheduler, as it removes 
+   * the need for a priority queue (and thus mutex locks).
+   * <p>
+   * go/static-stride-scheduler
+   * <p>
    *
    * <ul>
-   *   <li>In the first call, the deadlines are A=2 (1/0.5) and B=5 (1/0.2), so A is returned.
-   *   The deadline of A is updated to 4.
-   *   <li>Next, the remaining deadlines are A=4 and B=5, so A is returned. The deadline of A (2) is
-   *       updated to A=6.
-   *   <li>Remaining deadlines are A=6 and B=5, so B is returned. The deadline of B is updated with
-   *       with B=10.
-   *   <li>Remaining deadlines are A=6 and B=10, so A is returned. The deadline of A is updated with
-   *        A=8.
-   *   <li>Remaining deadlines are A=8 and B=10, so A is returned. The deadline of A is updated with
-   *       A=10.
-   *   <li>Remaining deadlines are A=10 and B=10, so A is returned. The deadline of A is updated
-   *      with A=12.
-   *   <li>Remaining deadlines are A=12 and B=10, so B is returned. The deadline of B is updated
-   *      with B=15.
-   *   <li>etc.
-   * </ul>
-   *
-   * <p>In short: the entry with the highest weight is preferred.
-   *
-   * <ul>
-   *   <li>add() - O(lg n)
-   *   <li>pick() - O(lg n)
-   * </ul>
-   *
+   *  <li>nextSequence() - O(1)
+   *  <li>pick() - O(n)
    */
   @VisibleForTesting
-  static final class EdfScheduler {
-    private final PriorityQueue<ObjectState> prioQueue;
-
-    /**
-     * Weights below this value will be upped to this minimum weight.
-     */
-    private static final double MINIMUM_WEIGHT = 0.0001;
-
-    private final Object lock = new Object();
+  static final class StaticStrideScheduler {
+    private final int[] scaledWeights;
+    private final int sizeDivisor;
+    private final AtomicInteger sequence;
+    private static final int K_MAX_WEIGHT = 0xFFFF;
```

Comment:
This value was chosen to have an array of shorts. I think we want that (`short[] scaledWeights`) since the precision of int really doesn't add much. Since Java doesn't have unsigned shorts, I think we should either make this value Short.MAX_VALUE (0x7FFF) or use `Short.toUnsignedInt()` when reading out of the array.

Since we only read and write to the array in one location, `Short.toUnsigned*()` makes a lot of sense to match the C impl and have the same amount of precision. If we interacted with the array frequently, using toUnsigned could be bug-prone.